Pangolin/Framework/Messaging/MessageQueue.cs

```diff
@@ -0,0 +1,199 @@
+        private const string _tableExistsStatement = @"SELECT OBJECT_ID('MessageQueue.{0}')";
+        private const string _insertMessageStatement = @"INSERT INTO [MessageQueue].{0} ([Priority],[DateCreated],[MessageBody]) VALUES (@Priority,@DateCreated,@MessageBody)";
+        private const string _readMessageStatement = @"WITH T AS (SELECT TOP 1 [Id],[Priority],[DateCreated],[MessageBody] FROM MessageQueue.{0} ORDER BY PRIORITY DESC, ID DESC) DELETE FROM T OUTPUT DELETED.ID, DELETED.Priority, DELETED.DateCreated, DELETED.MessageBody";        
+        private const string _peekMessageStatement = "SELECT TOP 1 [Id],[Priority],[DateCreated],[MessageBody] FROM MessageQueue.{0} ORDER BY PRIORITY DESC, ID DESC";
+        private const string _countMessagesStatement = "SELECT COUNT(*) FROM MessageQueue.{0}";
+        private const string _clearMessagesStatement = "DELETE FROM MessageQueue.{0}";
+
+        public string Name => _queueName;
+
+        /// <summary>
+        /// Constructs the object, builds the underlying table if it doesn't exist.
+        /// </summary>
+        /// <param name="Name">The name of the queue</param>
+        public MessageQueue(string databaseConnection, string queueName, ICache cache)
+        {
+            if (!IsQueueNameValid(queueName))
+            {
+                throw new ArgumentException("Queue name is not valid");
+            }
+            _databaseConnection = databaseConnection;
+            _queueName = queueName;
+            //TODO I don't know that this even needs to cache.  Users shouldn't be creating message queues frequently.  They will likely pass one around as a dependency.
+            if (!TableExists(cache))
+            {
+                CreateTable();
+            }
+        }
+
+        private bool IsQueueNameValid(string queueName)
+        {
+            return regex.IsMatch(queueName);
+        }
+
+        /// <summary>
+        /// Creates the table.
+        /// </summary>
+        private void CreateTable()
+        {
+            try
+            {
+                using (SqlConnection connection = new SqlConnection(_databaseConnection))
+                {
+                    connection.Open();
+                    using (SqlCommand command = new SqlCommand(string.Format(_createTableStatement, _queueName), connection))
+                    {
+                        command.ExecuteNonQuery();
+                    }
+                }
+            }
+            catch(SqlException ex)
+            {
+                if (ex.Number != 2714)
```

Comment:
Whats 2714?

Reply:
This is a good idea, I"m going to incorporate it.